src/server/svr_jobfunc.c

```diff
@@ -5723,3 +5724,1613 @@ svr_avljob_oper(job *pjob, int delkey)
+							       extra_res) == NULL) {
+								log_err(-1, id, "pbs_strcat failed");	
+								goto recreate_exec_vnode_exit;
+							}
+						}
+						sel_entry++;
+					}
+				}
+			} else {
+
+				if ((strcmp(pnode->nd_moms[0]->mi_host,
+						ms_fullhost) != 0) ||
+			     		(pnode->nd_moms[0]->mi_port == \
+							ms_port)) {
+					if (f_entry > 0) { /* there's */
+				       		/* something put in */
+						/* previously */
+						strcat(deallocated_execvnode, "+");
+					}
+
+					if (((hasprn > 0) && (paren > 0)) ||
+				     	    ((hasprn == 0) && (paren == 0)) ) {
+						 /* at the beginning */
+						 /* of chunk for current host */
+						if (!parend1) {
+							strcat(deallocated_execvnode,
+									"(");
+							parend1 = 1;
+						}
+					}
+
+					if (!parend1) {
+						strcat(deallocated_execvnode, "(");
+						parend1 = 1;
+					}
+					strcat(deallocated_execvnode, chunk_buf);
+					f_entry++;
+
+					if (paren == 0) { /* have all chunks for */
+						  /* current host */
+
+						if (parend) {
+							strcat(new_exec_vnode, ")");
+							parend = 0;
+						}
+			
+						if (parend1) {
+							strcat(deallocated_execvnode,
+									 ")");
+							parend1 = 0;
+						}
+					}
+		
+				}
+
+				if (hasprn < 0) {
+					/* matched ')' in chunk, so need to */
+					/* balance the parenthesis */
+					if (parend) {
+						strcat(new_exec_vnode, ")");
+						parend = 0;
+					}
+					if (parend1) {
+						strcat(deallocated_execvnode, ")");
+						parend1 = 0;
+					}
+
+					buf_sum = resc_sum_values_action(RESC_SUM_GET_CLEAR,
+							NULL, NULL, NULL, err_msg, err_msg_sz);
+
+					if (buf_sum == NULL) {
+						log_event(PBSEVENT_DEBUG2, PBS_EVENTCLASS_JOB, LOG_DEBUG,
+							id, err_msg);
+						goto recreate_exec_vnode_exit;
+					}
+
+					if (buf_sum[0] != '\0') {
+						extra_res = return_missing_resources(chunk3,
+								res_in_exec_vnode);
+				
+						if (sel_entry > 0) {
+							/* there's already previous */
+							/* select/schedselect entry */
+							if (pbs_strcat(
+								&new_select,
+								&ns_malloced,
+								"+") == NULL) {
+								log_err(-1, id, "pbs_strcat failed");	
+								goto recreate_exec_vnode_exit;
+							}
+						}
+						if (pbs_strcat( &new_select,
+							&ns_malloced,
+								"1") == NULL) {
+							log_err(-1, id,
+							"pbs_strcat failed");
+							goto recreate_exec_vnode_exit;
+						}
+						if (pbs_strcat( &new_select,
+							&ns_malloced,
+								buf_sum) == NULL) {
+							log_err(-1, id,
+							"pbs_strcat failed");
+							goto recreate_exec_vnode_exit;
+						}
+						if ((extra_res != NULL) && (extra_res[0] != '\0')) {
+							if (pbs_strcat(
+								&new_select,
+								&ns_malloced,
+								":") == NULL) {
+								log_err(-1, id, "pbs_strcat failed");	
+								goto recreate_exec_vnode_exit;
+							}
+							if (pbs_strcat(
+								&new_select,
+								&ns_malloced,
+								extra_res) == NULL) {
+								log_err(-1, id, "pbs_strcat failed");	
```

Comment:
Please combine the multiple line pbs_strcat() calls into fewer lines.  They are hard to read at 4 lines each.